soh/soh/Enhancements/randomizer/draw.cpp

```diff
@@ -131,78 +178,162 @@ extern "C" void Randomizer_DrawBossKey(PlayState* play, GetItemEntry* getItemEnt
     gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx, (char*)__FILE__, __LINE__),
               G_MTX_MODELVIEW | G_MTX_LOAD);
 
-    if (color_slot == 5 && isColoredKeysEnabled) { // Ganon's Boss Key
-        gDPSetGrayscaleColor(POLY_OPA_DISP++, 80, 80, 80, 255);
-        gSPGrayscale(POLY_OPA_DISP++, true);
-    }
-
-    gSPDisplayList(POLY_OPA_DISP++, (Gfx*)gGiBossKeyDL);
-
-    if (color_slot == 5 && isColoredKeysEnabled) { // Ganon's Boss Key
-        gSPGrayscale(POLY_OPA_DISP++, false);
+    Color_RGB8 keyColor = { 255, 255, 0 };
+    //Supposed to use CVAR_COSMETIC but I can't figure out the syntax
+    keyColor = CVarGetColor24(("gCosmetics.Key." + CvarValue[slot] + "BossBase.Value").c_str(), keyColor);
+    
+    if (isCustomKeysEnabled){
+        gDPSetEnvColor(POLY_OPA_DISP++, keyColor.r, keyColor.g, keyColor.b, 255);
+        gSPDisplayList(POLY_OPA_DISP++, (Gfx*)gBossKeyCustomDL);
+    } else {
+        if (CVarGetInteger(("gCosmetics.Key." + CvarValue[slot] + "BossBase.Changed").c_str(), false)){
+            gDPSetGrayscaleColor(POLY_OPA_DISP++, keyColor.r, keyColor.g, keyColor.b, 255);
+            gSPGrayscale(POLY_OPA_DISP++, true);
+            gSPDisplayList(POLY_OPA_DISP++, (Gfx*)gGiBossKeyDL);
+            gSPGrayscale(POLY_OPA_DISP++, false);
+        } else {
+            gSPDisplayList(POLY_OPA_DISP++, (Gfx*)gGiBossKeyDL);
+        }
     }
```

Comment:
I'm not entirely sure what's going on here. Why is it using the .Changed parameter to apply the Grayscale color? What happens if you change the value when a boss key isn't rendered at that point?

Reply:
Gotcha, my apologies. Then this definitely makes sense 👍